File: 392-is-subsequence/leochang/392.cpp
class Solution {
public:
    bool isSubsequence(string s, string t) {
        int s_index = 0;
        if (s.length() == 0) return true;
        if (s.length() != 0 && t.length() == 0) return false;
        
        for (int i = 0; i < t.length(); i++) {
            if (s[s_index] == t[i]) s_index++;
            if (s_index == s.length()) return true;
        }
        
        return false;
    }
};

File: 392-is-subsequence/ken/392.py
class Solution(object):
    def isSubsequence(self, s, t):
        """
        :type s: str
        :type t: str
        :rtype: bool
        """

        if len(s) == 0:
            return True

        j = 0

        for i in xrange(len(t)):
            if t[i] == s[j]:
                j += 1
                if j == len(s):
                    break

        return j == len(s)
